woocommerce/woocommerce
# Delayed account creation: Use Block Hooks to inject block into the Order Confirmation template

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This is based on https://github.com/woocommerce/woocommerce/pull/51236. Please review https://github.com/woocommerce/woocommerce/pull/51236 first.

Since the Account Creation Block is shown/hidden based on a global setting it needs to exist in the template. This is a strong use case for Block Hooks which this PR introduces.

The Account Creation block is hooked in after the order confirmation summary block. Default content is injected. This is compatible with already existing custom templates!

Block locking prevents removal. The global setting to enable/disable this feature will hide the block if desired. 

Closes #51322



### How to test the changes in this Pull Request:

**This block is behind an experimental flag. If you are testing on a production version of WooCommerce this block should not be available (yet).**

These instructions are for testing the development build:

1. Before using this PR, go ahead and reset the template to default under Appearance > Editor > Templates, and then save a custom version of the template. Add some text somewhere so you know its custom.
2. Use this PR
3. Under Settings > Accounts and Privacy, enable creating an account after checkout
4. Use a different browser. As a guest user, using a new billing address, place an order.
5. Confirm order confirmation shows the create account form!
6. Back in admin, disable creating an account after checkout.
7. Refresh your confirmation page. Create account block is now hidden.
8. Back in admin, Appearance > Editor > Templates, reset changes to the template so its back as default.
9. Repeat tests 3-7.
10. Back in admin, Appearance > Editor > Templates, test you can edit the template, see the Account Creation block, and it cannot be removed because its locked.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
